src/internal/operators/delayWhen.ts

```diff
@@ -140,12 +140,16 @@ class DelayWhenSubscriber<T, R> extends OuterSubscriber<T, R> {
   private tryDelay(delayNotifier: Observable<any>, value: T): void {
     const notifierSubscription = subscribeToResult(this, delayNotifier, value);
 
-    if (notifierSubscription && !notifierSubscription.closed) {
-      this.add(notifierSubscription);
-      this.delayNotifierSubscriptions.push(notifierSubscription);
+    if (notifierSubscription) {
+      if (notifierSubscription.closed) {
```

Comment:
I guess a problem with this is that this is also true if the selector errors synchronously(?). Is there a better way to check this?